pkg/provider/aws/knowledgebase/edges.go

```diff
@@ -235,11 +288,233 @@ var AwsKB = knowledgebase.EdgeKB{
+			integration := source.(*resources.ApiIntegration)
+			// This isnt an instance of an expanded path, rather an existing edge so ignore
+			if integration.Name != "" {
+				return nil
+			}
+			function := dest.(*resources.LambdaFunction)
+			restApi, ok := data.Source.(*resources.RestApi)
+			refs := core.DedupeAnnotationKeys(append(function.ConstructsRef, restApi.ConstructsRef...))
+			if !ok {
+				return fmt.Errorf("source of lambda to api integration expansion must be a rest api resource")
+			}
+			if len(data.Routes) == 0 {
+				return fmt.Errorf("there are no routes to expand the edge for lambda to api integration")
+			}
+
+			for _, route := range data.Routes {
+				var err error
+				integration, err = core.CreateResource[*resources.ApiIntegration](dag, resources.ApiIntegrationCreateParams{
+					AppName:    data.AppName,
+					Refs:       refs,
+					Name:       route.Path,
+					ApiName:    restApi.Name,
+					HttpMethod: strings.ToUpper(string(route.Verb)),
+				})
+				if err != nil {
+					return err
+				}
+				integration.IntegrationHttpMethod = "POST"
+				integration.Type = "AWS_PROXY"
+				integration.Uri = core.IaCValue{Resource: function, Property: resources.LAMBDA_INTEGRATION_URI_IAC_VALUE}
+				segments := strings.Split(route.Path, "/")
+				segment := segments[len(segments)-1]
+				methodRequestParams := map[string]bool{}
+				integrationRequestParams := map[string]string{}
+				if strings.Contains(segment, ":") {
+					// We strip the pathParam of the : and * characters (which signal path parameters or wildcard routes) to be able to inject them into our method and integration request parameters
+					pathParam := fmt.Sprintf("request.path.%s", segment)
+					pathParam = strings.ReplaceAll(pathParam, ":", "")
+					pathParam = strings.ReplaceAll(pathParam, "*", "")
+					methodRequestParams[fmt.Sprintf("method.%s", pathParam)] = true
+					integrationRequestParams[fmt.Sprintf("integration.%s", pathParam)] = fmt.Sprintf("method.%s", pathParam)
+				}
+				integration.RequestParameters = integrationRequestParams
+				integration.Method.RequestParameters = methodRequestParams
+
+				permission, err := core.CreateResource[*resources.LambdaPermission](dag, resources.LambdaPermissionCreateParams{
+					Name: fmt.Sprintf("%s-%s", function.Name, integration.RestApi.Id()),
+					Refs: refs,
+				})
+				if err != nil {
+					return err
+				}
+				permission.Function = function
+				permission.Principal = "apigateway.amazonaws.com"
+				permission.Action = "lambda:InvokeFunction"
+				permission.Source = core.IaCValue{Resource: integration.RestApi, Property: resources.API_GATEWAY_EXECUTION_CHILD_RESOURCES_IAC_VALUE}
+				for _, res := range dag.GetUpstreamResources(restApi) {
+					switch resource := res.(type) {
+					case *resources.ApiDeployment:
+						dag.AddDependency(resource, integration.Method)
+						dag.AddDependency(resource, integration)
```

Comment:
yeah you might be right, i might be able to remove that, i think we just had it before and i copied over the deps

Reply:
oh we do need it for the deployment to work. If there arent any methods in the depends on of the deployment then we fail because it thinks its deploying nothing.